eng/pipelines/templates/stages/source-build-stages.yml

```diff
@@ -0,0 +1,242 @@
+          vmrBranch: ${{ variables.VmrBranch }}
+          targetArchitecture: ${{ leg.targetArchitecture }}
+          ${{ if eq(leg.targetArchitecture, 'arm64') }}:
+            pool: ${{ parameters.pool_LinuxArm64 }}
+          ${{ else }}:
+            pool: ${{ parameters.pool_Linux }}
+          buildSourceOnly: true
+          runOnline: ${{ leg.runOnline }}
+          runTests: true
+          enablePoison: ${{ leg.enablePoison }}
+          skipBuild: true
+          extraProperties: >
+            /p:DotNetSourceOnlyTestOnly=true
+            /p:ExtraRestoreSourcePath=$(Pipeline.Workspace)/msft-pkgs
+            /p:RestoreAdditionalProjectSources=$(Pipeline.Workspace)/msft-pkgs
+          
+          testInitSteps:
+          # Ensure the artifacts staging directory exists so that even if no files get placed there, it won't fail
+          # when publishing the artifacts.
+          - script: mkdir -p $(artifactsStagingDir)
+            displayName: Create Artifacts Staging Directory
+          - ${{ if eq(leg.distro, 'alpine') }}:
+            - template: ../steps/download-artifacts.yml
+              parameters:
+                artifactDescription: Microsoft Linux (musl) Packages
+                artifactName: AzureLinux_x64_Cross_Alpine_${{ leg.targetArchitecture }}_Artifacts
+                downloadFilePatterns: packages/Release/Shipping/**
+                copyDestination: $(Pipeline.Workspace)/msft-pkgs
+                flattenDirs: true
+          - ${{ else }}:
+            - template: ../steps/download-artifacts.yml
+              parameters:
+                artifactDescription: Microsoft Linux Packages
+                artifactName: AzureLinux_x64_Cross_${{ leg.targetArchitecture }}_Artifacts
+                downloadFilePatterns: packages/Release/Shipping/**
+                copyDestination: $(Pipeline.Workspace)/msft-pkgs
+                flattenDirs: true
+          - template: ../steps/download-artifacts.yml
+            parameters:
+              artifactDescription: Microsoft WASM Packages
+              artifactName: Browser_Shortstack_wasm_Artifacts
+              downloadFilePatterns: packages/Release/Shipping/**
+              copyDestination: $(Pipeline.Workspace)/msft-pkgs
+              flattenDirs: true
+          - template: ../steps/download-artifacts.yml
+            parameters:
+              artifactDescription: Microsoft Windows Packages
+              artifactName: Windows_x64_Artifacts
+              downloadFilePatterns: packages/Release/Shipping/**
+              copyDestination: $(Pipeline.Workspace)/msft-pkgs
+              flattenDirs: true
+          - template: ../steps/download-artifacts.yml
+            parameters:
+              artifactDescription: Source Build Assets
+              downloadFilePatterns: assets/Release/**/*.tar.gz
+              copyDestination: $(Build.SourcesDirectory)/artifacts
+              ${{ if eq(leg.distro, 'centos') }}:
```

Comment:
Did you consider passing in the variable distro name (e.g. variables.centOSStreamName) in instead of using the distro parameter?  This would yield a couple of benefits.

1. In practice it would provide some safety in using the predefined variable versus free form entry.  It would help ensure the distro is handled and eliminate human error typing in a value.
2. In cased like this it would eliminate the if..else logic as you could define a single value using the distro name parameter.  It would also help simplify the vmr-build.yml distro parameters.

Reply:
We can't pass in a variable value from the source-build.yml file because of how AzDO works. That's the whole reason for using this magic string.

Reply:
Thoughts on capturing this is a comment within the template?  I can see this coming up again.

Reply:
It's described here: https://github.com/dotnet/sdk/pull/48703/files#diff-11eee42bd01ce3be309ca482d06e6a19ae73bac5d44ba519a5a17b3a72641ad9R30-R35

Reply:
Insert joke here about developers not reading documentation.